test/unit/app/renderer/components/ledgerTableTest.js

```diff
@@ -0,0 +1,544 @@
+      faviconURL: ''
+    })
+  ])
+}
+
+describe('LedgerTable component', function () {
+  before(function () {
+    mockery.enable({
+      warnOnReplace: false,
+      warnOnUnregistered: false,
+      useCleanCache: true
+    })
+
+    mockery.registerMock('electron', fakeElectron)
+    mockery.registerMock('../../../../js/settings', fakeSettings)
+    fakeSettings.mockReturnValue = false
+    window.chrome = fakeElectron
+    mockery.registerMock('../../../../extensions/brave/img/ledger/verified_green_icon.svg')
+    mockery.registerMock('../../../../extensions/brave/img/ledger/verified_white_icon.svg')
+    mockery.registerMock('../../../../extensions/brave/img/ledger/icon_remove.svg')
+    mockery.registerMock('../../../../extensions/brave/img/ledger/icon_pin.svg')
+    LedgerTable = require('../../../../../app/renderer/components/preferences/payment/ledgerTable')
+  })
+
+  after(function () {
+    mockery.disable()
+  })
+
+  it('only non pinned tabs', function () {
+    const siteSettings = Immutable.Map([
+      [
+        'https?://times.com',
+        Immutable.Map({
+          ledgerPayments: true,
+          ledgerPinPercentage: 0
+        })
+      ]
+    ])
+
+    const synopsis = Immutable.List([
+      Immutable.Map({
+        site: 'times.com',
+        verified: false,
+        views: 2,
+        pinPercentage: 0,
+        secondsSpent: 60,
+        minutesSpent: 0,
+        hoursSpent: 0,
+        daysSpent: 0,
+        score: 3.6513249998816057,
+        publisherURL: 'http://times.com',
+        duration: 59963,
+        faviconURL: ''
+      })
+    ])
+
+    const wrapper = mount(
+      <LedgerTable
+        ledgerData={Immutable.Map({synopsis: synopsis})}
+        settings={Immutable.Map()}
+        onChangeSetting={function () {}}
+        siteSettings={siteSettings}
+      />
+    )
+    assert.equal(wrapper.find('[data-tbody-index="0"] [data-test-id="siteName"]').length, 0, '0 pinned')
+    assert.equal(wrapper.find('[data-tbody-index="1"] [data-test-id="siteName"]').length, 1, '1 unpinned')
+  })
+
+  it('two pinned tabs, 3 unpinned (show all button, 1 hidden unpinned)', function () {
+    const wrapper = mount(
+      <LedgerTable
+        ledgerData={Immutable.Map({synopsis: fivePublishers.synopsis})}
+        settings={Immutable.Map()}
+        onChangeSetting={function () {}}
+        siteSettings={fivePublishers.siteSettings}
+      />
+    )
+    assert.equal(wrapper.find('[data-tbody-index="0"] [data-test-id="siteName"]').length, 2, '2 pinned')
+    assert.equal(wrapper.find('[data-tbody-index="1"] [data-test-id="siteName"]').length, 2, '2 unpinned, 1 hidden')
+    assert.equal(wrapper.find('[data-l10n-id="showAll"]').length, 1, 'show all button visible')
+  })
+
+  it('two pinned tabs, 3 unpinned (hide less button, all visible)', function () {
+    const wrapper = mount(
+      <LedgerTable
+        ledgerData={Immutable.Map({synopsis: fivePublishers.synopsis})}
+        settings={Immutable.Map({
```

Comment:
@bsclifton This test is failing, because settings is not found in the `getSettings` in `ledgerTable.js` file https://github.com/brave/browser-laptop/pull/7642/files#diff-e168dd994d1e4de3f340af83899a20f5R247.

Problem is that this if is failing https://github.com/brave/browser-laptop/blob/master/js/settings.js#L58. More specific this is `false` `settingsCollection.constructor === Immutable.Map`. Which is strange, becuase as you can see I am passing `settings` prop as an `Immutable.Map`.

Any idea what is the problem?